### PR TITLE
Remove mayavi dependency

### PR DESCRIPTION
**Pull request by vlukes**
This PR removes the dependency on _Mayavi_.

- `make` - passed
- `./test_install.py` - 0 failed, 203 passed, 657 warnings in 294.19 seconds

It resolves issues #108, #687.

### CODE REVIEW

**Comment by vlukes** [comment]
I only checked `*.*` so I missed `INSTALL` and `sfepy-run`.

```
doc/primer.rst
doc/using_salome.rst
```
are obsolete parts of the documentation. Do you really think that it is necessary to update post-processing instructions there?

**Comment by rc** [comment]
`doc/using_salome.rst` is probably obsolete, I did not checked its functionality a long time (maybe never :)). Anyway, it needs just deleting two lines... We could remove it or mark it obsolete later.

However `doc/primer.rst` is a nice tutorial that should work, I update it regularly.

**Comment by rc** [comment]
LGTM!

**Comment by rc** [comment]
Ha, I have found more files to edit (`git grep -l sfepy.postprocess.viewer`):
- doc/developer_guide.rst - the postprocess package index
- doc/tutorial.rst
- sfepy/examples/diffusion/laplace_shifted_periodic.py
- sfepy/examples/linear_elasticity/its2D_interactive.py
- sfepy/examples/linear_elasticity/linear_elastic_interactive.py
- sfepy/examples/linear_elasticity/modal_analysis.py
- sfepy/examples/linear_elasticity/shell10x_cantilever_interactive.py

You can run
```
$ python3 script/sync_module_docs.py doc/src/ . -n
smd: removing unneeded rst files in "/home/eldaran/projects/sfepy-git/doc/src"...
smd:   remove: doc/src/sfepy/postprocess/sources.rst
smd:   remove: doc/src/sfepy/postprocess/utils.rst
smd:   remove: doc/src/sfepy/postprocess/viewer.rst
smd: ...done
smd: creating missing rst files in "/home/eldaran/projects/sfepy-git/doc/src"...
smd:   create: doc/src/homogen.rst
smd: ...done
```
to see new/removed modules. When you run it without `-n`, delete manually `doc/src/homogen.rst` :)

Also remove `sfepy/postprocess/domain_specific.py`.

File: sfepy/examples/linear_elasticity/shell10x_cantilever_interactive.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
   python sfepy/examples/linear_elasticity/shell10x_cantilever_interactive.py output -t bend -p
 
-Apply the twisting transformation to the beam domain coordinates, change number of cells, show the solution::
+Apply the twisting transformation to the beam domain coordinates, change number of cells::
 
   python sfepy/examples/linear_elasticity/shell10x_cantilever_interactive.py output -t twist -n 2,51,3 -s
```
Comment (rc) [inline sfepy/examples/linear_elasticity/shell10x_cantilever_interactive.py:50]: Remove `-s`

**Pull request merged**